### PR TITLE
New Netbanx Hosted Payments API Support

### PR DESCRIPTION
**Pull request by lukeholder**
Added support for the new Netbanx (Optimal Payments / Paysafe) Hosted Payments gateway API support.

Supports:

- Purchase (redirect)
- Authorize (redirect)
- Capture
- Refund

All new Netbanx customers are not allowed to use the old API anymore.

API Docs: https://developer.optimalpayments.com/en/documentation/hosted-payment-api/overview/

Tests for `capture()` and `refund()` (success and failure casses) still needed.

### CODE REVIEW

**Comment by lukeholder** [comment]
@greydnls once I get tests for refund and capture, as well as code formatting passing on travisCI, would you be the person to merge?

**Comment by delatbabel** [comment]
The official maintainer for this gateway is @borNfreee although any of the omnipay team can do the merge.

**Comment by lukeholder** [comment]
thanks @delatbabel will ping once ready.

**Comment by lukeholder** [comment]
This is my first time building a 'full' omnipay gateway, feedback welcome @borNfreee

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (borNfreee) [inline .idea/codeStyleSettings.xml]: I believe you should not commit any .idea/* files to this repository. Remove them please

File: src/Message/HostedAbstractResponse.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Omnipay\NetBanx\Message;
+
+abstract class HostedAbstractResponse extends \Omnipay\Common\Message\AbstractResponse
+{
+    public function getRedirectMethod()
+    {
+        return "GET";
```
Comment (borNfreee) [inline src/Message/HostedAbstractResponse.php]: use singlequotes here and in other files

File: src/Message/HostedAuthorizeRequest.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Omnipay\NetBanx\Message;
+
+class HostedAuthorizeRequest extends HostedPurchaseRequest
+{
+
```
Comment (borNfreee) [inline src/Message/HostedAuthorizeRequest.php]: remove extra empty line

File: src/Message/HostedCaptureRequest.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Omnipay\NetBanx\Message;
+
+class HostedCaptureRequest extends HostedAbstractRequest
+{
+
```
Comment (borNfreee) [inline src/Message/HostedCaptureRequest.php]: remove extra empty line

**Comment by borNfreee** [comment]
@lukeholder Hello, I'm not a maintainer of this NetBanx omnipay repository, I'm just a person who created it initialy without commit/merge access. 

In general, you PR looks good but please ping the real maintainer

**Comment by delatbabel** [comment]
Ping @barryvdh -- the omnipay repository has @borNfreee listed as the maintainer but I guess that should switch to one of the Omnipay team. Do you want to pick it up? NetBanx is EU so I guess closer to your sphere of influence than mine.

@borNfreee thanks for your work + code analysis.

**Comment by barryvdh** [comment]
@delatbabel To be honest, I never heard of this gateway and have no way of testing.

I'd rather make @borNfreee or @lukeholder a maintainer, as they both use it and clearly do have knowledge about this gateway. What do you think?

**Comment by lukeholder** [comment]
@barryvdh happy to put my hand up

**Comment by delatbabel** [comment]
> @barryvdh happy to put my hand up

Clearly showing insufficient reluctance.

**Comment by lukeholder** [comment]
switched to new branch name so I can composer install my fork, so new pull request:

https://github.com/thephpleague/omnipay-netbanx/pull/5

**Pull request closed without merging**